### ISSUE TITLE
Relative order of --coverage and --temp-directory command-line arguments shouldn't matter

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4657]
|<img alt="cgay's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| cgay reported|
|-|-|
|Trac ID|trac#4657|
|Type|defect|
|Created|2010-09-28 12:43:03Z|
|Branch|https://github.com/twisted/twisted/tree/trial-coverage-dir-4657|

Twisted 10.0.0

trial --coverage --temp-directory /tmp/foo

trial --temp-directory /tmp/foo --coverage

In the first example, coverage is stored in _trial_temp/.
In the second example, coverage is stored in /tmp/foo/.

This seems like it could indicate a basic problem in the way t.u.Options works, but I'll put it here for now because I don't know enough about how Options works.  It seems to me that all values should be stored before doing special handling via opt_*.

<details><summary>Searchable metadata</summary>

```
trac-id__4657 4657
type__defect defect
reporter__cgay cgay
priority__normal normal
milestone__ 
branch__branches_trial_coverage_dir_4657 branches/trial-coverage-dir-4657
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1285677783000000 1285677783000000
changetime__1287408181000000 1287408181000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [30096]) Branching to 'trial-coverage-dir-4657'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

I fixed this by removing the coverage directory calculation from `opt_coverage` entirely.  The value isn't needed until the run is complete anyway.  This guarantees the specified _temp-directory_ value will be respected.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set owner to @exarkun|
|-|-|

Please merge!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [30141]) Merge trial-coverage-dir-4657

Author: exarkun
Reviewer: therve
Fixes: [#4657](https://github.com/twisted/twisted/issues/4657)


Make trial's coverage option save the data to the same location
regardless of whether `--temp-directory` or `--coverage` is specified
first.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
